src/components/admin/Billing/graphs/DataByMonthGraph.tsx

```diff
@@ -0,0 +1,204 @@
+                xAxis: {
+                    type: 'category',
+                    data: months,
+                },
+                yAxis: {
+                    type: 'value',
+                    axisLabel: {
+                        formatter: '{value} GB',
+                    },
+                    splitLine: {
+                        lineStyle: {
+                            color: defaultOutlineColor[theme.palette.mode],
+                        },
+                    },
+                    minInterval: 0.001,
+                },
+                series: seriesConfig.map(({ data }) => ({
+                    type: 'line',
+                    data: data.map(([month, dataVolume]) => [
+                        month,
+                        (dataVolume / BYTES_PER_GB).toFixed(3),
+                    ]),
+                    markLine: {
+                        data: [{ yAxis: 10, name: 'GB\nFree' }],
+                        label: {
+                            color: theme.palette.text.primary,
+                            formatter: '{c} {b}',
+                            position: 'end',
+                        },
+                        lineStyle: {
+                            color: theme.palette.text.primary,
+                        },
+                        silent: true,
+                        symbol: 'none',
+                    },
+                    symbol: 'circle',
+                    symbolSize: 7,
+                })),
+                textStyle: {
+                    color: theme.palette.text.primary,
+                },
+                tooltip: {
+                    backgroundColor: paperBackground[theme.palette.mode],
+                    borderColor: defaultOutlineColor[theme.palette.mode],
+                    trigger: 'axis',
+                    textStyle: {
+                        color: theme.palette.text.primary,
+                        fontWeight: 'normal',
+                    },
+                    formatter: (tooltipConfigs: any[]) => {
+                        if (tooltipConfigs.length > 0) {
+                            const config = tooltipConfigs[0];
+
+                            const dataVolume = formatDataVolumeForDisplay(
+                                seriesConfig,
+                                config
+                            );
+
+                            const tooltipTitle =
+                                billingHistory
+                                    .map(({ date }) =>
+                                        intl.formatDate(date, {
+                                            month: 'short',
+                                            year: 'numeric',
+                                        })
+                                    )
+                                    .find((date) =>
+                                        date.includes(config.axisValueLabel)
+                                    ) ?? config.axisValueLabel;
```

Comment:
Future work:
This code runs a lot. We should prepare the tooltip in advance in a manner that allows us to quickly grab it with the config passed in.